### PR TITLE
Feature/tao 7233/remove controller contructor

### PR DESCRIPTION
**Pull request by siwane**

### CODE REVIEW

File: actions/DataCreation.php [hunk 1]
```diff
@@ -29,34 +29,30 @@
...
  * @license GPLv2 http://www.opensource.org/licenses/gpl-2.0.php
- *         
+ *
  */
 class DataCreation extends \tao_actions_Main
 {
-
-    public function __construct()
-    {
-        parent::__construct();
```
Comment (ionutpad) [inline actions/DataCreation.php]: The parent it's calling `$this->setIframeHeaders();` are you sure you don't need that in the actions here?

File: actions/DataCreation.php [hunk 2]
```diff
@@ -29,34 +29,30 @@
...
-    }
-
     /**
      * Work in progress
      */
     public function createItems()
     {
+        $this->defaultData();
```
Comment (ionutpad) [inline actions/DataCreation.php]: No need for `$this->defaultData()` it's just an echo return, no template

File: actions/ItemTools.php [hunk 3]
```diff
@@ -1,68 +1,71 @@
...
         $this->setData('jsonPackage', htmlentities($json));
         $this->setView('ItemTools/viewPackage.tpl');
     }
-    
-    public function getAsset() {
-        $item = new \core_kernel_classes_Resource($this->getRequestParameter('id'));
+
+    public function getAsset()
+    {
+        $item = $this->getResource($this->getRequestParameter('id'));
```
Comment (ionutpad) [inline actions/ItemTools.php]: Better to be safe and load the constants here as well

File: actions/UserDebug.php [hunk 4]
```diff
@@ -31,26 +31,19 @@
...
-	/**
-	 * initialize the services
-	 */
-	public function __construct(){
-		parent::__construct();
-		$this->userService = \tao_models_classes_UserService::singleton();
-	}
+	protected function getUserService()
+    {
+        return \tao_models_classes_UserService::singleton();
```
Comment (ionutpad) [inline actions/UserDebug.php]: This is deprecated use the `ServiceLocator`

File: actions/DataCreation.php (hunk 1 above)
Comment (siwane) [reply actions/DataCreation.php]: It is called automatically

**Pull request merged**